src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerArgumentBinder.cs

```diff
@@ -118,15 +118,16 @@ public class ControllerArgumentBinder : IControllerActionArgumentBinder
                 parameter.BindingInfo,
                 parameter.Name);
 
-            var modelBindingResult = await operationContext.ModelBinder.BindModelAsync(modelBindingContext);
-            if (modelBindingResult.IsModelSet)
+            await operationContext.ModelBinder.BindModelAsync(modelBindingContext);
+            var modelBindingResult = modelBindingContext.Result;
+            if (modelBindingResult != null && modelBindingResult.Value.IsModelSet)
```

Comment:
Think `HasValue` is much more consistent w/ the rest of this repo than `!= null` (for `Nullable<T>`s).